react/features/chat/components/native/Chat.js

```diff
@@ -0,0 +1,196 @@
+ */
+class Chat extends Component<Props> {
+
+    /**
+       * Initializes a new Conference instance.
+       *
+       * @param {Object} props - The read-only properties with which the new
+       * instance is to be initialized.
+       */
+    constructor(props: Props) {
+        super(props);
+        this._onSend = this._onSend.bind(this);
+        this._renderMessage = this._renderMessage.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *x
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+
+        const messages = this.props._messages.map(this._renderMessage);
+
+        return (
+            <Modal
+                animationType = { 'slide' }
+                onRequestClose = { this.props._onToggleChat }
+                supportedOrientations = { [
+                    'landscape',
+                    'portrait'
+                ] }
+                transparent = { true }
+                visible = { this.props._isOpen }>
+                <Header>
+                    <BackButton onPress = { this.props._onToggleChat } />
+                    <HeaderLabel labelKey = 'chat.title' />
+                </Header>
+                <SafeAreaView style = { styles.modalBackdrop }>
+                    <GiftedChat
+                        messages = { messages.reverse() }
+                        onSend = { this._onSend }
+                        user = {{ _id: this.props._localParticipant.id }} />
+                </SafeAreaView>
+            </Modal>
+        );
+    }
+
+    _onSend: (Array<Object>) => void;
+
+    /**
+     * Called by {@code render} to create the chat div.
+     *
+     * @param {string} message - The chat message to display.
+     * @param {string} id - The chat message ID to use as a unique key.
+     * @returns {Array<ReactElement>}
+     */
+    _onSend([ message ]) {
+        this.props._onSendMessage(message.text);
+    }
+
+    _renderMessage: (Object) => void;
+
+    /**
+     * Called by {@code _onSubmitMessage} to create the chat message.
+     *
+     * @param {string} message - The chat message to display.
+     * @param {string} id - The chat message ID to use as a unique key.
+     * @returns {Array<ReactElement>}
+     */
+    _renderMessage(message: Object, id: string) {
+        return (
+            {
+                _id: id,
+                text: message.message,
+                createdAt: new Date(message.timestamp),
+                user: {
+                    _id: message.id,
+                    name: message.displayName
+                }
+            }
+        );
+    }
+}
+
+/**
+ * Maps redux actions to {@link Chat}'s React
+ * {@code Component} props.
+ *
+ * @param {Function} dispatch - The redux action {@code dispatch} function.
+ * @returns {{
+ *     _onToggleChat: Function,
+ *     _onSendMessage: Function
+ * }}
+ * @private
+ */
+function _mapDispatchToProps(dispatch: Dispatch<*>) {
```

Comment:
Are we using mapDispatchToProps now as a pattern? I forget why but it hasn't been commonly used. I'm for it.

Reply:
We do use it on mobile quite a bit.